### PR TITLE
More logging and tests for DelegateInvocationHandler

### PR DESCRIPTION
**Pull request by lutovich**

### CODE REVIEW

File: enterprise/ha/src/main/java/org/neo4j/kernel/ha/DelegateInvocationHandler.java [hunk 1]
```diff
@@ -137,7 +138,7 @@ private void log( String format, Object... arguments )
     {
         if ( log.isDebugEnabled() )
         {
-            log.debug( "[" + componentName + "]" + format, arguments );
+            log.debug( "[" + componentName + ", " + hashCode() + "] " + format, arguments );
```
Comment (thobe) [inline enterprise/ha/src/main/java/org/neo4j/kernel/ha/DelegateInvocationHandler.java:141]: Why is the hash code useful in the log?

Comment (thobe) [inline enterprise/ha/src/main/java/org/neo4j/kernel/ha/DelegateInvocationHandler.java:141]: At least add a comment about what purpose the `hashCode()` serves in the log message.

And if that purpose is to uniquely identify the instance, perhaps we should be explicit about it and use `System.identityHashCode( this )`.

File: enterprise/ha/src/test/java/org/neo4j/kernel/ha/DelegateInvocationHandlerTest.java [hunk 2]
```diff
@@ -42,8 +45,26 @@ public void shouldNotBeAbleToUseValueBeforeHardened() throws Exception
...
+    }
+
+    @Test
+    public void throwsWhenDelegateIsNotSet()
+    {
+        DelegateInvocationHandler<Value> handler = newDelegateInvocationHandler();
+
+        try
+        {
+            handler.invoke( null, Value.class.getDeclaredMethod( "get" ), new Object[0] );
```
Comment (thobe) [inline enterprise/ha/src/test/java/org/neo4j/kernel/ha/DelegateInvocationHandlerTest.java:62]: I know that the implementation doesn't use the proxy instance for anything, but passing null here still seems very odd.

**Comment by thobe** [comment]
Do we want some tests that verify that we log from the `DelegateInvocationHandler`?

File: enterprise/ha/src/main/java/org/neo4j/kernel/ha/DelegateInvocationHandler.java (hunk 1 above)
Comment (lutovich) [inline enterprise/ha/src/main/java/org/neo4j/kernel/ha/DelegateInvocationHandler.java:141]: Yes, hash code is needed to uniquely identify the instance. Will fix

**Comment by lutovich** [comment]
Tests for logging are probably not needed. Logging is only useful to investigate one particular problem. Intention is to remove logging then.

**Comment by lutovich** [comment]
@thobe since we've identified what the problem is this logging is no longer needed. I've created new PR with logging removed https://github.com/neo4j/neo4j/pull/5690. Same tests are present there and mentioned issue with `null` parameter fixed.

**Pull request closed without merging**